modules/social_features/social_comment/src/Plugin/GraphQL/DataProducer/QueryComments.php

```diff
@@ -77,13 +83,13 @@ class QueryComments extends EntityDataProducerPluginBase {
    * @return \Drupal\social_graphql\GraphQL\ConnectionInterface
    *   An entity connection with results and data about the paginated results.
    */
-  public function resolve(?NodeInterface $parent, ?int $first, ?string $after, ?int $last, ?string $before, bool $reverse, string $sortKey, RefinableCacheableDependencyInterface $metadata) {
-    if (is_string($parent)) {
-      $nodes = $this->entityTypeManager->getStorage('node')->loadByProperties(['uuid' => $parent]);
+  public function resolve(?NodeInterface $parent, ?string $bundle, ?int $first, ?string $after, ?int $last, ?string $before, bool $reverse, string $sortKey, RefinableCacheableDependencyInterface $metadata) {
+    if ($parent) {
+      $nodes = $this->entityTypeManager->getStorage('node')->loadByProperties(['uuid' => $parent->uuid()]);
       $parent = reset($nodes);
     }
 
-    $query_helper = new CommentQueryHelper($sortKey, $this->entityTypeManager, $this->graphqlEntityBuffer, $parent);
+    $query_helper = new CommentQueryHelper($sortKey, $this->entityTypeManager, $this->graphqlEntityBuffer, $parent, $bundle);
```

Comment:
We added a `bundle` since produced query can get comments from different types, so it was possible to see an incorrect comment list on the topic (e.g. we were able to see a comment from a group post as a topic comment in GraphQL response).